### PR TITLE
STUD-17 feat tasks

### PR DESCRIPTION
**Pull request by Ahmedelsa3eed**
## using TaskDAO
the admin have the ability to **add**, **edit**, and **remove** a task to specific course

### CODE REVIEW

File: Student-Unity/src/main/java/csed/swe/studentunity/Logic/RegistrationService.java [hunk 1]
```diff
@@ -15,6 +15,7 @@ public class RegistrationService {
     public String addUser(RegistrationRequest request) {
         // check if email is already in use, if not: add user to database and send verification email
         if(userService.getUser(request.getEmail()).isEmpty()) {
+            System.out.println(request);
```
Comment (AhmedElemary57) [inline Student-Unity/src/main/java/csed/swe/studentunity/Logic/RegistrationService.java:18]: pleas do not use printing statements it is better to use a logger if you want to show something in the backend.
I think that you were checking some thing put be careful as we do not want to pull those print lines.

Comment (AhmedElemary57) [reply Student-Unity/src/main/java/csed/swe/studentunity/Logic/RegistrationService.java:18]: I advise you to use the sonar Lint extension in IntelliJ as it warning you that there is a code smell in that line :) 
that is just to not accumulate all code smells at the end of the milestone.

File: Student-Unity/src/main/java/csed/swe/studentunity/API/TaskAPI.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+public class TaskAPI {
+
+    private final TaskService taskService;
+
+    public TaskAPI(TaskService taskService) {
+        this.taskService = taskService;
+    }
+
+    @PostMapping("/addTask")
+    public ResponseEntity<Task> addTask(@RequestBody Task task) {
```
Comment (AhmedElemary57) [inline Student-Unity/src/main/java/csed/swe/studentunity/API/TaskAPI.java:20]: You said the admin has the ability to do those operations but where is the check that the request is sent by the admin?
I advise you to use an interceptor to check that the request is sent by the admin.

**Review by AhmedElemary57 — CHANGES_REQUESTED** [review]
You did a great job I just want to have more clarifications on that PR please answer my comments.

File: Student-Unity/src/main/java/csed/swe/studentunity/Logic/RegistrationService.java (hunk 1 above)
Comment (Ahmedelsa3eed) [reply Student-Unity/src/main/java/csed/swe/studentunity/Logic/RegistrationService.java:18]: Ok well i forgot to remove that line actually.
I will try out sonarLint

File: Student-Unity/src/main/java/csed/swe/studentunity/API/TaskAPI.java (hunk 2 above)
Comment (Ahmedelsa3eed) [reply Student-Unity/src/main/java/csed/swe/studentunity/API/TaskAPI.java:20]: really i forgot to check that only the admin who can add a task
i will use the interceptor

**Review by AhmedElemary57 — APPROVED** [review]
Great work 😊

**Pull request merged**